src/element/mutateElement.ts

```diff
@@ -9,45 +10,35 @@ type ElementUpdate<TElement extends ExcalidrawElement> = Omit<
 // This function tracks updates of text elements for the purposes for collaboration.
 // The version is used to compare updates when more than one user is working in
 // the same drawing.
-export function mutateElement(
-  element: ExcalidrawElement,
-  updates?: ElementUpdate<ExcalidrawElement>,
+export function mutateElement<TElement extends ExcalidrawElement>(
+  element: TElement,
+  updates: ElementUpdate<TElement>,
 ) {
-  if (updates) {
-    Object.assign(element, updates);
+  const mutableElement = element as any;
+
+  for (const key in updates) {
+    const value = (updates as any)[key];
+    if (typeof value !== "undefined") {
+      mutableElement[key] = value;
+    }
   }
-  (element as any).version++;
-  (element as any).versionNonce = randomSeed();
-}
 
-export function newElementWith(
-  element: ExcalidrawElement,
-  updates: ElementUpdate<ExcalidrawElement>,
-): ExcalidrawElement {
-  return {
-    ...element,
-    ...updates,
-    version: element.version + 1,
-    versionNonce: randomSeed(),
-  };
-}
+  if (
+    typeof updates.height !== "undefined" ||
+    typeof updates.width !== "undefined" ||
+    typeof updates.points !== "undefined"
+  ) {
```

Comment:
Is this list exhaustive? What about font for text?

Reply:
Changing the font size seems to be working fine.

Reply:
i am not sure that this list is exhaustive. but i think it will change in the near future anyway. if it's not broken let's ship it